File: 90 DWT 3/EmailViewController.m
//
//  EmailViewController.m
//  90 DWT 1
//
//  Created by Jared Grant on 7/12/12.
//  Copyright (c) 2012 g-rantsoftware.com. All rights reserved.
//

#import "EmailViewController.h"

@interface EmailViewController ()

@end

@implementation EmailViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view.
    
    [self configureViewForIOSVersion];
    self.defaultEmail.text = ((SettingsNavController *)self.parentViewController).emailAddress;
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}

- (IBAction)hideKeyboard:(id)sender {
    [self.defaultEmail resignFirstResponder];
}

- (IBAction)saveEmail:(id)sender {
    // Get path to documents directory
    NSString *docDir = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES)[0];
    NSString *defaultEmailFile = nil;
    defaultEmailFile = [docDir stringByAppendingPathComponent:@"Default Email.out"];
    
    // Create the file
    [[NSFileManager defaultManager] createFileAtPath:defaultEmailFile contents:nil attributes:nil];
    
    // Write file to documents directory
    NSFileHandle *fileHandle = [NSFileHandle fileHandleForWritingAtPath:defaultEmailFile];
    [fileHandle writeData:[self.defaultEmail.text dataUsingEncoding:NSUTF8StringEncoding]];
    [fileHandle closeFile];
    
    ((SettingsNavController *)self.parentViewController).emailAddress = self.defaultEmail.text;
    
    // Clear the text field and show the placeholder text.
    self.defaultEmail.placeholder = self.defaultEmail.text;
    self.defaultEmail.text = @"";
}

- (void)configureViewForIOSVersion {
    
    // Colors
    UIColor *lightGrey = [UIColor colorWithRed:234/255.0f green:234/255.0f blue:234/255.0f alpha:1.0f];
    UIColor* blueColor = [UIColor colorWithRed:0/255.0f green:122/255.0f blue:255/255.0f alpha:1.0f];
    
    // Apply Text Colors
    self.emailLabel.textColor = blueColor;
    
    // Apply Background Colors
    self.view.backgroundColor = lightGrey;
    
    // Apply Keyboard Color
    self.defaultEmail.keyboardAppearance = UIKeyboardAppearanceDark;
}
@end

File: 90 DWT 3/EmailViewController.swift
//
//  EmailViewController.swift
//  90 DWT 3
//
//  Created by Grant, Jared on 9/21/16.
//  Copyright © 2016 Grant, Jared. All rights reserved.
//

import UIKit
import CoreData

class EmailViewController: UIViewController {

    // **********
    let debug = 0
    // **********
    
    @IBOutlet weak var emailLabel: UILabel!
    @IBOutlet weak var defaultEmail: UITextField!
    
    @IBAction func hideKeyboard(_ sender: AnyObject) {
        
        self.defaultEmail.resignFirstResponder()
    }
    
    @IBAction func saveEmail(_ sender: UIBarButtonItem) {
        
        // Fetch defaultEmail data.
        let request = NSFetchRequest<NSFetchRequestResult>( entityName: "Email")
        let sortDate = NSSortDescriptor( key: "date", ascending: true)
        request.sortDescriptors = [sortDate]
        
        do {
            if let emailObjects = try CoreDataHelper.shared().context.fetch(request) as? [Email] {
                
                if debug == 1 {
                    
                    print("emailObjects.count = \(emailObjects.count)")
                }
                
                switch emailObjects.count {
                case 0:
                    
                    // No Matches.  Create new record and save.
                    let insertEmailInfo = NSEntityDescription.insertNewObject(forEntityName: "Email", into: CoreDataHelper.shared().context) as! Email
                    
                    insertEmailInfo.defaultEmail = self.defaultEmail.text
                    insertEmailInfo.date = Date() as NSDate?
                    
                default:
                    
                    // There is a default email address.
                    emailObjects.last?.defaultEmail = self.defaultEmail.text
                }
                
                CoreDataHelper.shared().backgroundSaveContext()
                self.defaultEmail.placeholder = self.defaultEmail.text
                self.defaultEmail.text = ""
            }
        } catch { print(" ERROR executing a fetch request: \( error)") }
    }
}
